lib/identity_cache/query_api.rb

```diff
@@ -132,7 +143,12 @@ def instrumented_record_from_coder(coder) #:nodoc:
 
       def record_from_coder(coder) #:nodoc:
         if coder
-          klass = coder[:class].constantize
+          begin
+            klass = coder[:class].constantize
+          rescue NameError
+            raise InexistentAssociationError
+          end
```

Comment:
I decided to use this constant to avoid conflicts with other similar errors. This can be useful when developing the lib, as a missing class would be handled as a miss and would be way harder to find the issue.

Reply:
hmm, it is reasonable to specify a class in this context for sure, what I'm wondering here is about the inheritance situation, why is`class InexistentAssociationError < StandardError; end` not a `NameError`  subclass? I could make an OOP argument for that, but also there is the more pragmatic we probably still want the exception when bubbling up to tell us what constant it was looking for, for ease of debugging.

WDYT?

Reply:
Makes complete sense!